streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -1271,4 +1279,81 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
 
         builder.addGraphNode(streamsGraphNode, processNode);
     }
+
+    @Override
+    public KTable<K, V> toTable() {
+        return toTable(NamedInternal.empty());
+    }
+
+    @Override
+    public KTable<K, V> toTable(Named named) {
+        return doToTable(named, null);
+    }
+
+    @Override
+    public <VR> KTable<K, VR> toTable(Materialized<K, VR, KeyValueStore<Bytes, byte[]>> materialized) {
+        return toTable(NamedInternal.empty(), materialized);
+    }
+
+    @Override
+    public <VR> KTable<K, VR> toTable(Named named,
+                                      Materialized<K, VR, KeyValueStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(materialized, "materialized can't be null");
+
+        final MaterializedInternal<K, VR, KeyValueStore<Bytes, byte[]>> materializedInternal =
+            new MaterializedInternal<>(materialized);
+        return doToTable(named, materializedInternal);
+    }
+
+    private <VR> KTable<K, VR> doToTable(Named named,
+                                         MaterializedInternal<K, VR, KeyValueStore<Bytes, byte[]>> materializedInternal) {
+        Objects.requireNonNull(named, "named can't be null");
+        final String name = new NamedInternal(named).orElseGenerateWithPrefix(builder, TO_KTABLE_NAME);
+        final Serde<K> keySerde;
+        final Serde<VR> valueSerde;
+        final String queryableStoreName;
+        final String storeName;
+        final StoreBuilder<TimestampedKeyValueStore<K, VR>> storeBuilder;
+
+        if (materializedInternal != null) {
+            keySerde = materializedInternal.keySerde() != null ? materializedInternal.keySerde() : this.keySerde;
+            valueSerde = materializedInternal.valueSerde();
```

Comment:
Why can't we use `this.valueSerde` is `.valueSerde() == null` similar to `keySerde?